### PR TITLE
Refactor heading hold modes

### PR DESCRIPTION
**Pull request by digitalentity**
- [x] Remove `HEADING LOCK` mode
- [x] Rename `MAG` to `HEADING HOLD`
- [x] Allow `HEADING HOLD` mode to operate without compass

### CODE REVIEW

**Comment by oleost** [comment]
@digitalentity How is this for rudder less airplanes like flying wings?

File: src/main/sensors/compass.h [hunk 1]
```diff
@@ -57,7 +53,7 @@ typedef struct compassConfig_s {
     sensor_align_e mag_align;               // mag alignment
     uint8_t mag_hardware;                   // Which mag hardware to use on boards with more than one device
     flightDynamicsTrims_t magZero;
-    uint8_t mag_hold_rate_limit;            // Maximum rotation rate MAG_HOLD mode can feed to yaw rate PID controller
+    uint8_t __dummy_1;                      // Maximum rotation rate MAG_HOLD mode can feed to yaw rate PID controller
```
Comment (oleost) [inline src/main/sensors/compass.h:56]: Is this supposed to be _dummy_1 ?

Comment (digitalentity) [reply src/main/sensors/compass.h:56]: It is supposed to be `dummy_1`, will fix.

**Comment by digitalentity** [comment]
This doesn't work for rudderless airplanes yet - let's do it one step at a time.

**Pull request merged**